testsuite/MDAnalysisTests/coordinates/test_fhiaims.py

```diff
@@ -23,32 +23,159 @@
+                     "wrong time of the frame")
+
+    def test_bad_input1(self, tmpdir):
+        fname = str(tmpdir.mkdir("FHIAIMSReader").join('bad_input1.in'))
+        with open(fname, 'w') as fd:
+            fd.write('lattice_vector 1.0 0.0 0.0\n')
+            fd.write('lattice_vector 0.0 1.0 0.0\n')
+            fd.write('atom 0.1 0.1 0.1 H')
+        try:
+            u = mda.Universe(fname)
+        except Exception as e:
+            if 'Found partial periodicity' not in str(e):
+                raise ValueError(
+                    "FHIAIMSReader did not detect a missing lattice_vector")
+
+    def test_bad_input2(self, tmpdir):
+        fname = str(tmpdir.mkdir("FHIAIMSReader").join('bad_input2.in'))
+        with open(fname, 'w') as fd:
+            fd.write('garbage')
+        try:
+            u = mda.Universe(fname)
+        except Exception as e:
+            if 'Non-conforming line' not in str(e):
+                raise ValueError(
+                    "FHIAIMSReader did not detect a non-conforming line")
+
 
+class TestFHIAIMSWriter(BaseWriterTest):
+    prec = 6
+    ext = ".in"
 
-@pytest.fixture
-def FHIAIMS_U():
-    return mda.Universe(FHIAIMS)
+    @pytest.fixture(scope='class')
+    def ref(self):
+        return RefFHIAIMS()
 
+    @pytest.fixture
+    def universe(self):
+        return mda.Universe(FHIAIMS, FHIAIMS)
 
-def test_fhiaims_positions(FHIAIMS_U):
-    # first particle
-    ts = FHIAIMS_U.trajectory[0]
-    assert_almost_equal(FHIAIMS_U.atoms.positions[0],
-                        np.asarray([6.861735000000,   2.103823000000,  37.753513000000], dtype=np.float32))
+    @pytest.fixture
+    def universe2(self):
+        return mda.Universe(FHIAIMS)
 
+    @pytest.fixture
+    def outfile(self, tmpdir):
+        return str(tmpdir.mkdir("FHIAIMSWriter").join('primitive-fhiaims-writer' + self.ext))
 
-def test_fhiaims_n_frames(FHIAIMS_U):
-    assert len(FHIAIMS_U.trajectory) == 1
+    def test_writer(self, universe, outfile):
+        """Test writing from a single frame FHIAIMS file to a FHIAIMS file.
+        """
+        universe.atoms.write(outfile)
+        u = mda.Universe(FHIAIMS, outfile)
+        assert_almost_equal(u.atoms.positions,
+                            universe.atoms.positions, self.prec,
+                            err_msg="Writing FHIAIMS file with FHIAIMSWriter "
+                                    "does not reproduce original coordinates")
 
+    def test_writer_no_atom_names(self, u_no_names, outfile):
+        u_no_names.atoms.write(outfile)
+        u = mda.Universe(outfile)
+        expected = np.array(['X'] * u_no_names.atoms.n_atoms)
+        assert_equal(u.atoms.names, expected)
 
-def test_fhiaims_dimensions(FHIAIMS_U):
-    ts = FHIAIMS_U.trajectory[0]
-    assert_almost_equal(ts.dimensions,
-                        np.asarray([18.600000000000, 18.600000000000,  55.800000000000, 90., 90., 90.], dtype=np.float32))
+    def test_no_container(self, tmpdir, ref):
```

Comment:
Can probably be removed as it is in `BaseWriterTest`

Reply:
Also, the current test does not `assert` anything.

Reply:
removed